### ISSUE TITLE
Urdu Translation Progress

### ISSUE THREAD

**Issue by tesseralis** [reactjs/ur.reactjs.org#1]
## Maintainer List

* @jabranr
* @armujahid
* @ahmadawais 
* @abdullah1908 

## For New Translators

To translate a page:

1. Check that no one else has claimed your page in the checklist and comments below.
2. Comment below with the name of the page you would like to translate. **Please take only one page at a time**.
3. Clone this repo, translate your page, and submit a pull request!

Before contributing, read the glossary and style guide (once they exist) to understand how to translate various technical and React-specific terms.

Please be prompt with your translations! If you find find that you can't commit any more, let the maintainers know so they can assign the page to someone else.

## For Maintainers

When someone volunteers, edit this issue with the username of the volunteer, and with the PR. Ex:

* [ ] Home Page (@tesseralis) #1
* [x] RTL Support (#4)

When PRs are merged, make sure to mark that page as completed!

## Core Pages

To do before releasing as an "official" translation. **Please translate these pages first.**

* [ ] Home Page (@Sanaullahbugti) #9 
* [ ] Tutorial (@shahzad31) #10 

### Basics

* [ ] Hello World (@jabranr) #2 
* [ ] Introducing JSX (@arximughal) #5 
* [ ] Rendering Elements (@owais4321) #11 
* [ ] Components and Props (@viveksharmaui) #13 
* [ ] State and Lifecycle (@aaftab1441) #15 
* [x] Handling Events (#22)
* [ ] Conditional Rendering (@asheerrizvi) (#26) 
* [ ] Lists and Keys
* [ ] Forms
* [ ] Lifting State Up
* [ ] Composition vs Inheritance
* [ ] Thinking in React

### API Reference

* [ ] React
* [ ] React.Component
* [ ] ReactDOM
* [ ] ReactDOMServer
* [ ] DOM Elements
* [ ] SyntheticEvent
* [ ] Test Utilities
* [ ] Shallow Renderer
* [ ] Test Renderer
* [ ] JS Environment Requirements
* [ ] Glossary

### Navigation

These are the navigation links that appears in the sidebar. Possibly wait until
the corresponding sections are translated to do these.

* [ ] tutorial/nav.yml
* [ ] docs/nav.yml

## Next Steps

These are the next most important translations, ordered by importance:

### Installation

* [ ] Getting Started
* [ ] Add React to a Website
* [ ] Create a New React App
* [ ] CDN Links

### Hooks

Everyone is hooked on hooks! People are really excited for these and it would be great for them to be translated.

  * [ ] Introducing Hooks
  * [ ] Hooks at a Glance
  * [ ] Using the State Hook
  * [ ] Using the Effect Hook
  * [ ] Rules of Hooks
  * [ ] Building Your Own Hooks
  * [ ] Hooks API Reference
  * [ ] Hooks FAQ

### Warnings

These are the pages that you get when you click the links in the console (e.g. https://reactjs.org/warnings/dont-call-proptypes.html). People tend to search these, so **please don't translate the error message itself**.
 
* [ ] dont-call-proptypes
* [ ] invalid-aria-prop
* [ ] invalid-hook-call-warning
* [ ] legacy-factories
* [ ] refs-must-have-owner
* [ ] special-props
* [ ] unknown-prop

### Advanced Guides

* [ ] Accessibility
* [ ] Code-Splitting
* [ ] Context
* [ ] Error Boundaries
* [ ] Forwarding Refs
* [ ] Fragments
* [ ] Higher-Order Components
* [ ] Integrating with Other Libraries
* [ ] JSX In Depth
* [ ] Optimizing Performance
* [ ] Portals
* [ ] React Without ES6
* [ ] React Without JSX
* [ ] Reconciliation
* [ ] Refs and the DOM
* [ ] Render Props
* [ ] Static Type Checking
* [ ] Strict Mode
* [ ] Typechecking With PropTypes
* [ ] Uncontrolled Components
* [ ] Web Components

### FAQ

* [ ] APIs and AJAX
* [ ] Babel, JSX, and Build Steps
* [ ] Passing Functions to Components
* [ ] Component State
* [ ] Styling and CSS
* [ ] File Structure
* [ ] Versioning Policy
* [ ] Virtual DOM and Internals

### Contributing

Priority: Low

* [ ] How to Contribute
* [ ] Codebase Overview
* [ ] Implementation Notes
* [ ] Design Principles

### Components

Components in `src/components` that have some text in them.

* [ ] CodeEditor
* [ ] CodeExample
* [ ] ErrorDecoder
* [ ] LayoutFooter
* [ ] LayoutHeader
* [ ] MarkdownPage

## Additional Translations

These are not the primary translation targets.

* [ ] Blog

### Community
* [ ]  Courses (@mhkasif) #21

**Comment by ahmadawais** [comment]
Howdy, @tesseralis!
I'd love to join the Reactjs Org to translate React's resource in Urdu.

I'm one of the [Locale Managers](https://make.wordpress.org/polyglots/teams/?locale=ur) at WordPress for Urdu and helped arrange several meetups for consistently [translating WordPress in Urdu](https://ahmadawais.com/urdu-wordpress/).

Can you invite me (username `ahmadawais`) in so I can help manage the local community along with my dear friend @jabranr and hopefully along with @saqibameen @ahmadbilalme. Would also love to conduct a small meetup in March about this.

Looking forward, peace! ✌️

**Comment by jabranr** [comment]
@ahmadawais welcome! 🎉 I've added you as member. You also need a small PR in this repository (https://github.com/reactjs/reactjs.org-translation) to update list of maintainers for Urdu. Refer to their readme for more info.

I am currently working on some basic guidelines, hopefully be able to share soon. For that purpose, I have created an issue #3 for discussions and everyone's input in this regard.

**Comment by saqibameen** [comment]
Thanks for the mention @ahmadawais. Would definitely love to contribute to this effort. 🤞

**Comment by jabranr** [comment]
This is great! When ready, please create an issue to claim a page from above list so it can be assigned.

**Comment by ahmadawais** [comment]
Thanks a lot @jabranr looking forward to contributing.

**Comment by abdullah1908** [comment]
@jabranr as mentioned on the [PR](https://github.com/reactjs/reactjs.org-translation/pull/60#issuecomment-468391083), I am also interested to maintain the translation :) You can add me as well in the maintainers list.
Thank you

**Comment by jabranr** [comment]
@abdullah1908 I would suggest the same as @armujahid had; to create a PR here in this repository and claim a page for translation. You will have to claim a page for yourself by looking into what suites you best. It is difficult for us to assign a page to you.

Also do you want to discuss your idea of adding Urdu strings in #3?

**Comment by abdullah1908** [comment]
Thanks @jabranr :relaxed:
Sure, I would love to discuss& share ideas.

**Comment by arximughal** [comment]
Hi @jabranr , I'd love to translate **Introducing JSX**.

**Comment by arximughal** [comment]
@jabranr Here's an issue https://github.com/reactjs/ur.reactjs.org/issues/5, you can assign it to me.

**Comment by tesseralis** [comment]
Hey folks! How are things going here? @jabranr @arximughal how are your translations going?

**Comment by jabranr** [comment]
Hi @tesseralis, thanks for checking. I had been quite busy lately so have not done much. I aim to get back on couple of things I started earlier over this weekend. 

What is your status @arximughal?

**Comment by arximughal** [comment]
Hi @tesseralis @jabranr, sorry for checking in late. I am actually a little busy with some personal stuff. I am moving to a new place (yay!). But the translation is going well. I'll submit the first version this weekend for you guys' review.

**Comment by sajid086** [comment]
Hi @tesseralis @jabranr I would love to contribute `React Docs` translation in to `Urdu` please add me. I have commented in the slack channel about `React Urdu` translation.

**Comment by Sajid-Ali** [comment]
Hi @tesseralis  @jabranr  I would love to contribute React Docs translation in to Urdu please add me. I have commented in the slack channel about React Urdu translation.

**Comment by jabranr** [comment]
Hi @Sajid-Ali, welcome and thank you for your interest in translating React docs. First of all you need to choose one page that you want to start translating from. To do this, [create an issue](https://github.com/reactjs/ur.reactjs.org/issues/new) in this repository ([here is an example](https://github.com/reactjs/ur.reactjs.org/issues/5)) and then one of us can assign you that page in the list given above. 

In order to contribute translation, you can always start by forking this repository and making a pull request back to it. If you have any question, please feel free to ask.

**Comment by jabranr** [comment]
@tesseralis quick question. Are we able to use `netlify` app to preview builds in submitted PRs? I am asking because it seems to be already installed in this repo.

**Comment by tesseralis** [comment]
@jabranr sure!

**Comment by jabranr** [comment]
I have setup a PR commit hook with netlify so it will make a preview-able build for each PR submitted in this repository. The most recent build based on #6 is available at https://gifted-bhabha-a2b6d0.netlify.com

Beware that URLs will be dynamic and unique for each build. Those should be available inside the PR going forward.

**Comment by armujahid** [comment]
https://ur.reactjs.org is now up.

**Comment by shahzad31** [comment]
I would like to contribute to this, i can pick up tutorials page i have create an issue , kindly assign it to so that i don't create a conflicting progress.

**Comment by jabranr** [comment]
Sure @shahzad31. welcome aboard!

**Comment by owais4321** [comment]
Hello @jabranr  I would like to contribute to this, I want to take Rendering Elements page for translation thanks

**Comment by jabranr** [comment]
@owais4321 Welcome. Sure. You need to create an issue with a title describing the page you wish to work on. Let us know here and we then assign it to you in list above.

**Comment by owais4321** [comment]
Thanks Jabran for having me on board  I have opened up an issue for translating rendering element page
________________________________

**Comment by aaftab1441** [comment]
Hello @jabranr i would like to contribute to this effort and i have created an issue #10 please assign it to me.

**Comment by jabranr** [comment]
@aaftab1441 Welcome! Did you mean #15 ?

**Comment by viveksharmaui** [comment]
@jabranr I want to build react local community for ur.reactjs.org is it possible ?  because i observed that we need a community for this to be translated in urdu.

**Comment by mhkasif** [comment]
I would like to contribute in translating files to urdu. 
I converted community/courses.md file already and looking forward to convert more files.

Do i have my own choice to select?


Thanks.

**Comment by jabranr** [comment]
@mhkasif  Welcome. Yes you have your own choice. Just choose the one from the list above and shout here we those can be assigned to you. It is important so that there is no duplicate work in parallel.

**Comment by asheerrizvi** [comment]
Hello,

I would like to translate the "Handling Events". I have opened an issue for the same. 

Thanks!

**Comment by mumairofficial** [comment]
I am observing that there a couple of PR which are actually closed or inactive for more than a year. I think those assignments should be cleared out and available to others.

**Comment by asheerrizvi** [comment]
Would like to take one from #2, #5, #10, #11, #13 and #15 if they are inactive. Will that be possible @jabranr ?

**Comment by jabranr** [comment]
@asheerrizvi hard to say. People who have assigned these tasks are in better position to answer that. You are free to choose anything else from the rest of the list anyway.

**Comment by asheerrizvi** [comment]
@jabranr cool then, I will take "Conditional Rendering" and start it over the weekend.

**Comment by jabranr** [comment]
@asheerrizvi I have created a ticket #26 for you. Feel free to add more details to it.

**Comment by jabranr** [comment]
Hi everyone, today we can start working on translations for new beta docs. I have created a new issue with list of pages for these new docs. More information in https://github.com/reactjs/ur.reactjs.org/issues/59

See you all there!

**Comment by TheSlimCoder** [comment]
@jabranr we need more members for this translation work can I work on outreach will share it with other's otherwise we will close 2nd issue as well.

**Comment by jabranr** [comment]
@TheSlimCoder Agreed. Yes great. 

Regarding closing #2 and more, I think we will need a cleanup here so we start fresh for the new docs.

**Comment by TheSlimCoder** [comment]
@jabranr I will try uploading a video on youtube for our React Urdu docs initiative and will ask more and more open-source members to contribute within the repo. 

Thanks!

**Comment by jabranr** [comment]
@TheSlimCoder great and do share a link with us here.

**Comment by haseebhamza** [comment]
Hello, @jabranr 
I'd love to join the Reactjs Org on GitHub to translate React's resource in Urdu.

As I'm myself learning React for sometime now, I'd love to make my contribution towards translating React into my native language Urdu and make it accessible to everyone in my local community to learn React.

Can you invite me so I can start making my contributions (username @haseebhamza) 

Looking forward, peace 🤞

**Comment by zainzafar90** [comment]
Hello, @ahmadawais 

I want to contribute to the react community for people who want to read the docs in urdu. Can you invite me (username @zainzafar90)

Thanks

**Comment by jabranr** [comment]
Hi @haseebhamza and @zainzafar90,

Fantastic that you both want to join in the efforts. Even greater news is that you do not need any invites to take part. All you need to do is to create a new issue and assign a page to yourself and start cracking. Here is the list of pages to be translated:

https://github.com/reactjs/ur.reactjs.org/issues/59

Good luck! 👍🏼

**Comment by TheSlimCoder** [comment]
@jabranr here is a video link: https://www.youtube.com/watch?v=WfF682j2Kfo

**Comment by ahmadawais** [comment]
Best of luck @zainzafar90 🙌

**Comment by Estevamsl** [comment]
Because I want to stay on top of everything that goes on in this organization and of course, as a good dev, you have to know about the challenges of this incredible world of programming. Can you invite me (username @Estevamsl)

**Comment by gaearon** [comment]
It looks like this translation has stalled, and is now very outdated.
I'm going to archive this repo, but if someone wants to pick up, please start a new effort.
